src/network/networkd-ndisc.c

```diff
@@ -558,7 +558,7 @@ static int ndisc_router_process_default(Link *link, sd_ndisc_router *rt) {
                 if (!route_gw->protocol_set)
                         route_gw->protocol = RTPROT_RA;
                 if (!route_gw->pref_set)
-                        route->pref = preference;
+                        route_gw->pref = preference;
```

Comment:
Ouch.

Reply:
I think it'd be better to not use titles like "fix typo" for logic errors. Strictly speaking many nasty bugs are "typos" in some sense… Anyone looking just at the log will likely skip this commit as "not interesting".

Reply:
Thanks. Commit message is updated.